glean-core/src/ping/mod.rs

```diff
@@ -69,12 +70,44 @@ impl PingMaker {
         current_seq as usize
     }
 
+    /// Get the formatted start and end times for this ping and update for the next ping.
+    fn get_start_end_times(&self, glean: &Glean, storage_name: &str) -> (String, String) {
```

Comment:
What I would try to test here:

- do we get the Glean object start time if no data is stored?
- does the `start_time` get set to the `end_time`?

Reply:
1. That would require setting up the Glean object. in `glean-core/tests/*` that is easy to do, but there we can't (right now) access `start_time` of that object as it's not exposed (I can make it public, though I would not want to). Plus, it gets truncated to minutes and therefore `Glean::new().start_time()` formatted to minutes should be the same as `Local::now()`, therefore not testing much. (We don't [currently] have any mocking in place to change this)
	* We _could_ make `get_start_end_times` return the `DateTime<FixedOffset>` instead, and defer the string formatting.
2. Again, in all our tests `start_time` and `end_time` tend to be within the same minute already (we could get the full timestamp from the database and compare that, but to me that seems a bit far-reaching)

Reply:
I don't know if this is possible in Rust, but could we mock `Local::now()` to return whatever time we want?  Then we could simulate many minutes passing.

We do something similar for the `TimingManager` in glean-ac.

Reply:
Not easily. 
There are some mocking frameworks (I never used one), but I assume they all depend on mocking by using traits in places.

Reply:
I'd vote for moving on and filing a bug for adding this specific testing (and one for testing mocking frameworks.. :D )

Reply:
On it!